library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegatesManager.java

```diff
@@ -244,8 +244,16 @@ public int getItemViewType(@NonNull T items, int position) {
             return FALLBACK_DELEGATE_VIEW_TYPE;
         }
 
+
+        final String itemString;
```

Comment:
What about making `itemString` an `final String errorMessage` and then in the if check do somehting like this:

```java
final String errorMessage;
if (items instanceof List<?>) {
           final String itemString = ((List<?>) items).get(position).toString();
           errorMessage = "No AdapterDelegate added that matches item=" + itemString + " at position=" + position + " in data source";
}  else {
           errorMessage = "No AdapterDelegate added for item at position "+position+". Data set is "+items.toString();
}

throw new NullpointerException(errorMessage);

```